crates/service/src/service/indexer_service.rs

```diff
@@ -157,389 +140,373 @@ impl From<&BuildInfo> for IndexerServiceRelease {
     }
 }
 
-pub struct IndexerServiceOptions<I>
-where
-    I: IndexerServiceImpl + Sync + Send + 'static,
-{
-    pub service_impl: I,
+pub struct IndexerServiceOptions {
+    pub service_impl: SubgraphService,
     pub config: &'static Config,
     pub release: IndexerServiceRelease,
     pub url_namespace: &'static str,
-    pub extra_routes: Router<Arc<IndexerServiceState<I>>>,
+    pub extra_routes: Router<Arc<IndexerServiceState>>,
 }
 
-pub struct IndexerServiceState<I>
-where
-    I: IndexerServiceImpl + Sync + Send + 'static,
-{
+pub struct IndexerServiceState {
     pub config: Config,
     pub attestation_signers: Receiver<HashMap<Address, AttestationSigner>>,
     pub tap_manager: Manager<IndexerTapContext>,
-    pub service_impl: Arc<I>,
+    pub service_impl: SubgraphService,
 
     // tap
     pub escrow_accounts: Receiver<EscrowAccounts>,
     pub domain_separator: Eip712Domain,
 }
 
-pub struct IndexerService {}
-
-impl IndexerService {
-    pub async fn run<I>(options: IndexerServiceOptions<I>) -> Result<(), anyhow::Error>
-    where
-        I: IndexerServiceImpl + Sync + Send + 'static,
-    {
-        let http_client = reqwest::Client::builder()
-            .tcp_nodelay(true)
-            .timeout(Duration::from_secs(30))
-            .build()
-            .expect("Failed to init HTTP client");
-
-        let network_subgraph: &'static SubgraphClient = Box::leak(Box::new(
-            SubgraphClient::new(
-                http_client.clone(),
+pub async fn run(options: IndexerServiceOptions) -> Result<(), anyhow::Error> {
+    let http_client = reqwest::Client::builder()
+        .tcp_nodelay(true)
+        .timeout(Duration::from_secs(30))
```

Comment:
Even when it's obvious what the number means, I'm not a fan of magic numbers. Could you make 30 a const?